### PR TITLE
Users should not be able to prohibit their own login

### PR DESCRIPTION
**Pull request by a1012112796**
Because I think it's crazy and not reasonale , that if a user can prohibit themself to login. so suggest limit this choice on ui.

### CODE REVIEW

**Review by zeripath — APPROVED** [review]

**Comment by zeripath** [comment]
Coming up with a succinct title for this PR is remarkably difficult!

File: templates/admin/user/edit.tmpl [hunk 1]
```diff
@@ -74,7 +74,7 @@
 				<div class="inline field">
 					<div class="ui checkbox">
 						<label><strong>{{.i18n.Tr "admin.users.prohibit_login"}}</strong></label>
-						<input name="prohibit_login" type="checkbox" {{if .User.ProhibitLogin}}checked{{end}}>
+						<input name="prohibit_login" type="checkbox" {{if .User.ProhibitLogin}}checked{{end}} {{if (eq .User.ID .SignedUserID)}}disabled{{end}}>
```
Comment (zeripath) [inline templates/admin/user/edit.tmpl:77]: It might be helpful to enforce this in the code too.

Comment (a1012112796) [reply templates/admin/user/edit.tmpl:77]: done it, is it waht you like? Thanks

File: routers/admin/users.go [hunk 2]
```diff
@@ -243,7 +244,20 @@ func EditUserPost(ctx *context.Context, form auth.AdminEditUserForm) {
...
+
+	var doerID int64
+	var ok bool
+	if doerID, ok = ctx.Data["SignedUserID"].(int64); !ok {
+		ctx.ServerError("SignedUserID", fmt.Errorf("Can't get SignedUserID"))
+		return
+	}
+
+	// skip slef Prohibit Login
+	if doerID == u.ID {
```
Comment (6543) [inline routers/admin/users.go]:
```suggestion
	if ctx.User.ID == u.ID {
```

File: routers/admin/users.go [hunk 3]
```diff
@@ -243,7 +244,20 @@ func EditUserPost(ctx *context.Context, form auth.AdminEditUserForm) {
 	u.AllowGitHook = form.AllowGitHook
 	u.AllowImportLocal = form.AllowImportLocal
 	u.AllowCreateOrganization = form.AllowCreateOrganization
-	u.ProhibitLogin = form.ProhibitLogin
+
+	var doerID int64
```
Comment (6543) [inline routers/admin/users.go]: pleace delete line 248 to 254 use `ctx.User.ID`

**Review by 6543 — CHANGES_REQUESTED** [review]

File: routers/admin/users.go (hunk 3 above)
Comment (a1012112796) [reply routers/admin/users.go]: do it , Thanks for your suggestion. I forget this verb before

File: routers/admin/users.go [hunk 4]
```diff
@@ -243,7 +243,13 @@ func EditUserPost(ctx *context.Context, form auth.AdminEditUserForm) {
 	u.AllowGitHook = form.AllowGitHook
 	u.AllowImportLocal = form.AllowImportLocal
 	u.AllowCreateOrganization = form.AllowCreateOrganization
-	u.ProhibitLogin = form.ProhibitLogin
+
+	// skip slef Prohibit Login
```
Comment (guillep2k) [inline routers/admin/users.go]:
```suggestion
	// skip self Prohibit Login
```

**Review by guillep2k — APPROVED** [review]

**Review by 6543 — APPROVED** [review]
beside @guillep2k spell corect sugestion - lgtm

**Review by lafriks — APPROVED** [review]

**Comment by 6543** [comment]
ping lgtm

**Comment by techknowlogick** [comment]
ping LG-TM

**Pull request merged**